pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -339,3 +342,12 @@ func getOrCreatePVC(pr *v1alpha1.PipelineRun, c kubernetes.Interface) error {
 func isDone(status *v1alpha1.PipelineRunStatus) bool {
 	return !status.GetCondition(duckv1alpha1.ConditionSucceeded).IsUnknown()
 }
+
+func getTaskRunTimeout(pr *v1alpha1.PipelineRun, pipelineTimeout *metav1.Duration) *metav1.Time {
+	if pipelineTimeout == nil {
+		return nil
+	}
+	// If a timeout is specified for the PipelineRun, set the timeout for the task to be the time from now until the
+	// PipelineRun timeout would occur.
+	return &metav1.Time{Time: pr.Status.StartTime.Add(pipelineTimeout.Duration)}
+}
```

Comment:
i like that this function doesn't depend on the reconciler at all! and b/c of that, we could easily move it into a separate module, and even have a unit test or two for it :)

Reply:
May even be attached to `PipelineRun` struct :angel:

Reply:
aaaaaand I've gotten rid of this and just moved the equivalent logic within `createTaskRun`, but I can extract it again if you'd like?